src/coreclr/jit/inline.def

```diff
@@ -66,6 +66,7 @@ INLINE_OBSERVATION(TOO_MUCH_IL,               bool,   "too many il bytes",
 
 // ------ Callee Information -------
 
+INLINE_OBSERVATION(FOLDABLE_CALL,             bool,   "foldable call",                        INFORMATION, CALLEE)
```

Comment:
For some future PR, we could also track the intrinsic id here.

This would allow us to determine, at least the following interesting pieces:
* Number of intrinsic calls
* Number of hardware intrinsic calls
* Number of `Type.op_Equality` calls

There may be other interesting bits as well, but these are at least ones that have one or more of:
* significantly decreased cost (hwintrinsics are instructions not calls)
* additional optimizations (many intrinsics have CSE support)
* potential dead code elimination (type equality checks, particularly `where T : struct`)